apis/gasbuddy.js

```diff
@@ -0,0 +1,255 @@
+/**
+ * @file apis/gasbuddy.js
+ *
+ * @author fewieden
+ * @license MIT
+ *
+ * @see  https://github.com/fewieden/MMM-Fuel
+ */
+
+/**
+ * @external node-fetch
+ * @see https://www.npmjs.com/package/node-fetch
+ */
+const fetch = require('node-fetch');
+
+/**
+ * @external node-html-parser
+ * @see https://www.npmjs.com/package/node-html-parser
+ */
+const { parse } = require('node-html-parser');
+
+const BASE_URL = 'https://www.gasbuddy.com';
+const MAX_PAGE = 1;
+
+let config;
+
+/**
+ * @function getRequestPaths
+ * @description URL paths for fuel type to request data sorted by distance and by price.
+ *
+ * @param {string} type - Fuel type.
+ *
+ * @returns {string[]} URL paths for fuel type.
+ */
+function getRequestPaths(type) {
+    switch(type) {
```

Comment:
I'm not a big fan of switch statements. You could convert this to a map like in some other providers, e.g.

```js
// At the top of the file
const TYPES = {
    regular: 1,
    midgrade: 2,
    premium: 3,
    diesel: 4,
    e85: 5,
    unl88: 12
};

function getRequestPaths(type) {
    return `/home?search=${config.zip}&fuel=${TYPES[type]}&maxAge=0&method=all`;
}
```